pdbfixer/pdbfixer.py

```diff
@@ -108,6 +110,98 @@ def __init__(self, topology, positions, terminal=None):
             terminal = [False]*topology.getNumAtoms()
         self.terminal = terminal
 
+@dataclass
+class CCDAtomDefinition:
+    """
+    Description of an atom in a residue from the Chemical Component Dictionary (CCD).
+    """
+    atomName: str
+    symbol: str
+    leaving: bool
+    coords: mm.Vec3
+    charge: int
+    aromatic: bool
+
+@dataclass
+class CCDBondDefinition:
+    """
+    Description of a bond in a residue from the Chemical Component Dictionary (CCD).
+    """
+    atom1: str
+    atom2: str
+    order: Literal['SING', 'DOUB', 'TRIP', 'QUAD', 'AROM', 'DELO', 'PI', 'POLY']
+    aromatic: bool
+
+@dataclass
+class CCDResidueDefinition:
+    """
+    Description of a residue from the Chemical Component Dictionary (CCD).
+    """
+    residueName: str
+    smiles: Optional[str]
+    atoms: list[CCDAtomDefinition]
+    bonds: list[CCDBondDefinition]
+
+    @classmethod
+    def fromReader(cls, reader: PdbxReader) -> 'CCDResidueDefinition':
+        """
+        Create a CCDResidueDefinition by parsing a CCD CIF file.
+        """
+        data = []
+        reader.read(data)
+        block = data[0]
+
+        residueName = block.getObj('chem_comp').getValue("id")
+
+        descriptorsData = block.getObj("pdbx_chem_comp_descriptor")
+        typeCol = descriptorsData.getAttributeIndex("type")
+        smilesCol = descriptorsData.getAttributeIndex("descriptor")
+        smiles = None
+        for row in descriptorsData.getRowList():
+            if row[typeCol] in ["SMILES", "SMILES_CANONICAL"]:
+                smiles = row[smilesCol]
+                break
```

Comment:
You record the SMILES string but never use it anywhere.  What is it for?

Reply:
I think I was using it in a previous iteration. I'll remove it.